**Step 0**

In [1]:


import pandas as pd

# Download latest version

df = pd.read_csv("C:\\Users\\joshn\\Downloads\\hw-1-uc-2026\\train.csv")

**Step 1: Data**

In [3]:
# 1) How many data samples are included in the dataset?

df.shape[0]

# 2) Which problem will this dataset try to address?

# 3) What is the minimum value and the maximum value in the dataset?

df.min()
df.max()

# 4) How many features in each data samples?

df.shape[1]

# 5) Does the dataset have any missing information? E.g., missing features.

df.isnull().sum()

# 6) What is the label of this dataset?

# TARGET_deathRate

# 7) How many percent of data will you use for training, validation and testing?

train_percent = 0.70
validation_percent = 0.15
test_percent = 0.15

# 8) What kind of data pre-processing will you use for your training dataset?

# Print all results:

print("Number of data samples:", df.shape[0])
print("Minimum values:", df.min().to_dict())
print("Maximum values:", df.max().to_dict())
print("Number of features:", df.shape[1])
print("Missing values:", df.isnull().sum().to_dict())
print("Training set percentage:", train_percent)
print("Validation set percentage:", validation_percent)
print("Test set percentage:", test_percent)

Number of data samples: 2437
Minimum values: {'id': 2, 'avgAnnCount': 6.0, 'avgDeathsPerYear': 3, 'TARGET_deathRate': 59.7, 'incidenceRate': 201.3, 'medIncome': 23047, 'popEst2015': 827, 'povertyPercent': 3.2, 'studyPerCap': 0.0, 'binnedInc': '(34218.1, 37413.8]', 'MedianAge': 22.3, 'MedianAgeMale': 22.4, 'MedianAgeFemale': 22.3, 'Geography': 'Acadia Parish, Louisiana', 'AvgHouseholdSize': 0.0221, 'PercentMarried': 23.1, 'PctNoHS18_24': 0.0, 'PctHS18_24': 0.0, 'PctSomeCol18_24': 7.1, 'PctBachDeg18_24': 0.0, 'PctHS25_Over': 7.5, 'PctBachDeg25_Over': 2.5, 'PctEmployed16_Over': 17.6, 'PctUnemployed16_Over': 0.4, 'PctPrivateCoverage': 22.3, 'PctPrivateCoverageAlone': 15.7, 'PctEmpPrivCoverage': 13.5, 'PctPublicCoverage': 11.2, 'PctPublicCoverageAlone': 2.6, 'PctWhite': 10.1991551, 'PctBlack': 0.0, 'PctAsian': 0.0, 'PctOtherRace': 0.0, 'PctMarriedHouseholds': 22.99248989, 'BirthRate': 0.0}
Maximum values: {'id': 3046, 'avgAnnCount': 38150.0, 'avgDeathsPerYear': 14010, 'TARGET_deathRate': 36

**Analysis of Step 1 Results**
1) How many data samples are included in the dataset?

The dataset includes 2437 samples.

2) Which problem will this dataset try to address?




3) What is the minimum value and the maximum value in the dataset?

**Minimum values:** {'id': 2, 'avgAnnCount': 6.0, 'avgDeathsPerYear': 3, 'TARGET_deathRate': 59.7, 'incidenceRate': 201.3, 'medIncome': 23047, 'popEst2015': 827, 'povertyPercent': 3.2, 'studyPerCap': 0.0, 'binnedInc': '(34218.1, 37413.8]', 'MedianAge': 22.3, 'MedianAgeMale': 22.4, 'MedianAgeFemale': 22.3, 'Geography': 'Acadia Parish, Louisiana', 'AvgHouseholdSize': 0.0221, 'PercentMarried': 23.1, 'PctNoHS18_24': 0.0, 'PctHS18_24': 0.0, 'PctSomeCol18_24': 7.1, 'PctBachDeg18_24': 0.0, 'PctHS25_Over': 7.5, 'PctBachDeg25_Over': 2.5, 'PctEmployed16_Over': 17.6, 'PctUnemployed16_Over': 0.4, 'PctPrivateCoverage': 22.3, 'PctPrivateCoverageAlone': 15.7, 'PctEmpPrivCoverage': 13.5, 'PctPublicCoverage': 11.2, 'PctPublicCoverageAlone': 2.6, 'PctWhite': 10.1991551, 'PctBlack': 0.0, 'PctAsian': 0.0, 'PctOtherRace': 0.0, 'PctMarriedHouseholds': 22.99248989, 'BirthRate': 0.0}

**Maximum values:** {'id': 3046, 'avgAnnCount': 38150.0, 'avgDeathsPerYear': 14010, 'TARGET_deathRate': 362.8, 'incidenceRate': 1206.9, 'medIncome': 125635, 'popEst2015': 10170292, 'povertyPercent': 47.4, 'studyPerCap': 9762.308998, 'binnedInc': '[22640, 34218.1]', 'MedianAge': 624.0, 'MedianAgeMale': 64.7, 'MedianAgeFemale': 65.7, 'Geography': 'Zavala County, Texas', 'AvgHouseholdSize': 3.97, 'PercentMarried': 72.5, 'PctNoHS18_24': 62.7, 'PctHS18_24': 72.5, 'PctSomeCol18_24': 76.2, 'PctBachDeg18_24': 51.8, 'PctHS25_Over': 54.8, 'PctBachDeg25_Over': 42.2, 'PctEmployed16_Over': 80.1, 'PctUnemployed16_Over': 29.4, 'PctPrivateCoverage': 92.3, 'PctPrivateCoverageAlone': 78.9, 'PctEmpPrivCoverage': 70.7, 'PctPublicCoverage': 65.1, 'PctPublicCoverageAlone': 43.3, 'PctWhite': 100.0, 'PctBlack': 84.86602358, 'PctAsian': 42.61942454, 'PctOtherRace': 41.93025142, 'PctMarriedHouseholds': 71.70305677, 'BirthRate': 21.32616487}

4) How many features in each data samples?

Each data sample has 35 features.

5) Does the dataset have any missing information? E.g., missing features.

Missing values: {'id': 0, 'avgAnnCount': 0, 'avgDeathsPerYear': 0, 'TARGET_deathRate': 0, 'incidenceRate': 0, 'medIncome': 0, 'popEst2015': 0, 'povertyPercent': 0, 'studyPerCap': 0, 'binnedInc': 0, 'MedianAge': 0, 'MedianAgeMale': 0, 'MedianAgeFemale': 0, 'Geography': 0, 'AvgHouseholdSize': 0, 'PercentMarried': 0, 'PctNoHS18_24': 0, 'PctHS18_24': 0, 'PctSomeCol18_24': 1828, 'PctBachDeg18_24': 0, 'PctHS25_Over': 0, 'PctBachDeg25_Over': 0, 'PctEmployed16_Over': 123, 'PctUnemployed16_Over': 0, 'PctPrivateCoverage': 0, 'PctPrivateCoverageAlone': 480, 'PctEmpPrivCoverage': 0, 'PctPublicCoverage': 0, 'PctPublicCoverageAlone': 0, 'PctWhite': 0, 'PctBlack': 0, 'PctAsian': 0, 'PctOtherRace': 0, 'PctMarriedHouseholds': 0, 'BirthRate': 0}

PctSomeCol18_24 is missing 1,828 values
PctPrivateCoverageAlone is missing 480 values
PctEmployed16_Over is missing 123 values


6) What is the label of this dataset?

The datasets label is **TARGET_deathRate**

7) How many percent of data will you use for training, validation and testing?

From train.csv I will use:
    80% training
    20% validation
    0% testing (testing is coming from test.csv)

8) What kind of data pre-processing will you use for your training dataset?

I will drop the PctSomeCol18_24 column, and impute the PctEmployed16_Over and PctPrivateCoverageAlone columns with the median of the sets based on missingv values. I will drop columns like 'id' where no value is provided.

**Data cleanup**

In [ ]:
# Remove PctSomeCol18_24
df = df.drop(columns=['PctSomeCol18_24'])

# Fill PctEmployed16_Over missing values with medians
df['PctEmployed16_Over'] = df['PctEmployed16_Over'].fillna(
    df['PctEmployed16_Over'].median()
)

# Fill PctPrivateCoverageAlone missing values with medians
df['PctPrivateCoverageAlone'] = df['PctPrivateCoverageAlone'].fillna(
    df['PctPrivateCoverageAlone'].median()
)

**Step 2: Model**